src/ray/gcs/redis_context.cc

```diff
@@ -45,74 +46,95 @@ namespace ray {
 
 namespace gcs {
 
-// This is a global redis callback which will be registered for every
-// asynchronous redis call. It dispatches the appropriate callback
-// that was registered with the RedisCallbackManager.
-void GlobalRedisCallback(void *c, void *r, void *privdata) {
-  if (r == nullptr) {
-    return;
+CallbackReply::CallbackReply(redisReply *redisReply) {
+  RAY_CHECK(nullptr != redisReply);
+  RAY_CHECK(redisReply->type != REDIS_REPLY_ERROR) << "Got an error in redis reply:"
```

Comment:
```suggestion
  RAY_CHECK(redisReply->type != REDIS_REPLY_ERROR) << "Got an error in redis reply: "
```
and there are a few other places that need fix as well